node/src/components/linear_chain_sync.rs

```diff
@@ -378,18 +362,9 @@ impl<I: Clone + PartialEq + 'static> LinearChainSync<I> {
         }
     }
 
-    // Returns whether `block` can be considered the tip of the chain.
-    fn is_recent_block(&self, block: &Block) -> bool {
-        // Check if block was created "recently".
-        block.header().timestamp().elapsed() <= self.acceptable_drift
-    }
-
     // Returns whether we've just downloaded a switch block of a currently active era.
-    fn is_currently_active_era(&self, maybe_switch_block: &Option<Box<Block>>) -> bool {
-        match maybe_switch_block {
-            Some(switch_block) => switch_block.header().timestamp().elapsed() < self.shortest_era,
-            None => false,
-        }
+    fn is_currently_active_era(&self, block: &Block) -> bool {
+        block.header().timestamp().elapsed() < self.shortest_era
```

Comment:
It would be more accurate if we'd update `shortest_era` (and maybe call it `min_remaining_era_duration` or `earliest_era_end`?) with each incoming block (not only switch block) — or, probably, not make it a field at all, and compute it in each `is_currently_active_era` call:

It's the maximum of:
* minimum round length × number of _remaining_ blocks in this era, and
* switch block timestamp + minimum era duration.

Otherwise if e.g. the `shortest_era` is 2 hours and era 10 has just started, we would stop syncing at the end of era 9, needlessly initialize one more old era, and synchronize the whole era 9 protocol state.

Reply:
Or leave the previous code which would remember the last switch block and then check if the next block is inside a "current" era.

Reply:
Ah, of course. I thought this is equivalent, but obviously, it isn't (I'm tired :grimacing:). I'll have to change this a bit :+1:

Reply:
I've coded the `minimum round length × number of remaining blocks in this era` condition now, but I left out the other part. There's less state to store this way and I think it should work well enough - but if you disagree, I can add the other part, too.